src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionService.java

```diff
@@ -91,6 +91,41 @@ public interface SnowflakeConnectionService {
    */
   boolean isTableCompatible(String tableName);
 
+  /**
+   * Check whether the user has the role privilge to do schema evolution and whether the schema
+   * evolution option is enabled on the table
+   *
+   * @param tableName the name of the table
+   * @param role the role of the user
+   * @return whether schema evolution has the required permission to be performed
+   */
+  boolean hasSchemaEvolutionPermissionForRole(String tableName, String role);
+
+  /**
+   * Check if the role used by the connector has the permission to do schema evolution on the table
+   * through a cached map
+   *
+   * @param tableName the name of the table
+   * @return whether we have the permission to do schema evolution on the table
+   */
+  boolean hasSchemaEvolutionPermission(String tableName);
+
+  /**
+   * Alter table to add columns according to a map from columnNames to their types
+   *
+   * @param tableName the name of the table
+   * @param columnToType the mapping from the columnNames to their types
+   */
+  void appendColumns(String tableName, Map<String, String> columnToType);
```

Comment:
AppendColumn is a very generalized term, Can we be more specific? May be `appendColumnsToTable`?